autotx/utils/ethereum/networks.py

```diff
@@ -0,0 +1,137 @@
+from dataclasses import dataclass
+import json
+from typing import Union
+import requests
+from gnosis.eth import EthereumNetwork
+from web3 import Web3
+
+from autotx.utils.ethereum.constants import NATIVE_TOKEN_ADDRESS
+
+KLEROS_TOKENS_LIST = "https://t2crtokens.eth.link/"
+COINGECKO_TOKENS_LIST = "https://tokens.coingecko.com/uniswap/all.json"
+
+TOKENS_LIST = [KLEROS_TOKENS_LIST, COINGECKO_TOKENS_LIST]
+
+
+class NetworkInfo:
+    network: EthereumNetwork
+    transaction_service_url: str
+    tokens: dict[str, str]
+
+    def __init__(
+        self,
+        chain_id: int,
+    ):
+        self.network = EthereumNetwork(chain_id)
+        config = SUPPORTED_NETWORKS_CONFIGURATION_MAP.get(self.network)
+
+        if config == None:
+            raise Exception(f"Chain ID {chain_id} is not supported")
+
+        self.transaction_service_url = config.transaction_service_url
+        fetched_tokens = self.load_tokens()
+        self.tokens = (
+            self.sanitize_loaded_tokens(fetched_tokens, chain_id)
+            | config.default_tokens
+        )
+
+    def sanitize_loaded_tokens(
```

Comment:
I think this is more of a `filter_tokens_for_chain_id`

Reply:
There's no need for this and `load_tokens` to be a method instead of a solo function
You can also combine them into a function method as well

Reply:
good point - will update it so it is one function :smile: